alpenhorn/auto_import.py

```diff
@@ -48,52 +56,79 @@ def _import_file(node, root, file_path):
     - The file.
     - Information on the file, if it is of type "corr".
     - Indicates that the file exists on this node.
+
+    Parameters
+    ----------
+    node : storage.StorageNode
+        The node we are processing.
+    file_path : string
+        Path of file on the node to import. If is is an absolute path it must
+        be within the node root, otherwise is is assumed to be relative to
+        the node root.
     """
-    # global import_done
-    curr_done = True
-    fullpath = os.path.join(root, file_path)
-    log.debug("Considering %s for import." % fullpath)
-
-    # Skip the file if ch_master.py still has a lock on it.
-    dir_name = os.path.dirname(file_path)
-    base_name = os.path.basename(file_path)
-    if os.path.isfile(os.path.join(root, dir_name, ".%s.lock" % base_name)):
-        log.debug("Skipping \"%s\", which is locked by ch_master.py." % fullpath)
+
+    log.debug("Considering \"%s\" for import.", file_path)
+
+    # Occasionally the watchdog sends events on the node root directory itself. Skip these.
+    if file_path == node.root:
+        log.debug("Skipping import request on the node root itself \"%s\"", node.root)
+        return
+
+    # Ensure the path is an absolute path within the node
+    if os.path.isabs(file_path):
+        if not in_directory(file_path, node.root):
+            log.error("File \"%s\" was not an absolute path within the node \"%s\"", file_path, node.root)
+            return
+    else:
+        file_path = os.path.join(node.root, file_path)
+    abspath = os.path.normpath(file_path)
+
+    # Skip requests to import a directory. Again these are occasionally sent by the watchdog
```

Comment:
This could probably be the very first check in the function, to save us the effort of checking for inclusion in the root, normalizing, etc., when there is nothing that can be done with it.

Reply:
@jrs65 thoughts?

Reply:
Yeah, this one makes sense to change around. I'll change this one.

Reply:
Actually. I remember this now. The reason to not swap them around is that you might need to resolve the path relative to the node root before you can test that it's a directory or not.